CI: Cross-build images for ARM do not detect wayland-scanner

For example this can be seen in this [build log](https://github.com/Igalia/cog/runs/4140268316), apparently `wayland-scanner` exists but then some check fails:

```
-- Checking for module 'wayland-scanner'
-- Could NOT find WaylandScanner (missing: WAYLAND_SCANNER WAYLAND_SCANNER_VERSION WAYLAND_SCANNER_CODE_ARG) 
-- Checking for module 'wayland-protocols'
--   Found wayland-protocols, version 1.21
CMake Error at /home/runner/toolchain/sysroots/x86_64-pokysdk-linux/usr/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find WaylandProtocols (missing: WAYLAND_SCANNER)
Call Stack (most recent call first):
  /home/runner/toolchain/sysroots/x86_64-pokysdk-linux/usr/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindWaylandProtocols.cmake:202 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  platform/wayland/CMakeLists.txt:16 (find_package)
```

I wonder if this might be related to the wildcard I added in c8756bd074fdc936a722052e4775424aaecb32a0 😇

--- a/.github/toolchain.sh
+++ b/.github/toolchain.sh
@@ -3,7 +3,7 @@ set -eu -o pipefail
 
 # Info on available toolchains: https://wk-contrib.igalia.com/
 INSTALL_DIR=${1:-${HOME}/toolchain}
-BASEURL=https://wk-contrib.igalia.com/yocto/meta-perf-browser/browsers/nightly/sdk
+BASEURL=https://wk-contrib.igalia.com/yocto/meta-perf-browser/browsers/stable/sdk
 FILE=wandboard-mesa/browsers-glibc-x86_64-core-image-weston-browsers-cortexa9t2hf-neon-wandboard-mesa-toolchain-1.0.sh
 
 declare -r INSTALL_DIR BASEURL FILE

--- a/.github/toolchain.sh
+++ b/.github/toolchain.sh
@@ -1,5 +1,5 @@
 #! /bin/bash
-set -e
+set -eu -o pipefail
 
 # Info on available toolchains: https://wk-contrib.igalia.com/
 INSTALL_DIR=${1:-${HOME}/toolchain}
@@ -9,12 +9,43 @@ FILE=wandboard-mesa/browsers-glibc-x86_64-core-image-weston-browsers-cortexa9t2h
 declare -r INSTALL_DIR BASEURL FILE
 rm -f ~/toolchain.sh
 
-declare -a curl_opts=( --retry 3 -L )
+declare -a curl_opts=( --http1.1 --retry 3 -L -C - )
 if [[ -r ${INSTALL_DIR}/.installed ]] ; then
     curl_opts+=( --time-cond "${INSTALL_DIR}/.installed" )
 fi
 
-curl "${curl_opts[@]}" -o ~/toolchain.sh "${BASEURL}/${FILE}"
+declare -i tries=0
+
+function fetch_installer {
+    if [[ $(( tries++ )) -ge 10 ]] ; then
+        echo 'Maximum amount of retries reached, bailing out.' 1>&2
+        return 1
+    fi
+
+    local exit_code=0
+    if curl "${curl_opts[@]}" -o ~/toolchain.sh "${BASEURL}/${FILE}" ; then
+        return 0
+    else
+        exit_code=$?
+
+        if [[ ${exit_code} -eq 36 ]] ; then
+            echo "Bad resume (${exit_code}), restarting download from scratch..." 1>&2
+            rm -f ~/toolchain.sh
+        else
+            echo "Download error (${exit_code}), retrying download..." 1>&2
+        fi
+        local seconds=$(( RANDOM % 10 + 5 ))
+        printf 'Waiting... %i' "${seconds}"
+        while [[ $(( seconds-- )) -gt 0 ]] ; do
+            sleep 1
+            printf ' %i' "${seconds}"
+        done
+        echo '.'
+        fetch_installer
+    fi
+}
+
+fetch_installer
 
 if [[ -r ~/toolchain.sh ]] ; then
     echo 'Installing toolchain...'

--- a/.github/workflows/ci-cross.yml
+++ b/.github/workflows/ci-cross.yml
@@ -14,7 +14,7 @@ jobs:
       - name: Install Debian Packages
         run: |
           sudo apt update
-          sudo apt install -y cmake ninja-build
+          sudo apt install -y cmake ninja-build libwayland-bin
       - name: Cache
         uses: actions/cache@v2
         id: cache
@@ -31,6 +31,7 @@ jobs:
         run: |-
           . ~/toolchain/environment-setup-*-poky-linux-gnueabi
           cmake -GNinja -S. -Bbuild \
+            -DWAYLAND_SCANNER=/usr/bin/wayland-scanner \
             -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
             -DCOG_PLATFORM_FDO=ON \
             -DCOG_PLATFORM_DRM=ON

